src/input/type_record.cc

```diff
@@ -442,8 +445,6 @@ Record &Record::declare_key(const string &key, const KeyType &type,
         data_->declare_key(key, boost::shared_ptr<const TypeBase>(), &type, default_value, description);
     } else {
         // for Array, Double, Integer, we assume no static variables
-    	if (const_cast<KeyType *>( &type )->finish()) check_key_default_value(default_value, type, key);
```

Comment:
Method check_key_default_value is now called only in finish. In some cases call ( &type )->finish() causes finishing of some incomplete descendants of 'type'.